### PR TITLE
heffte: add sycl variant

### PR DESCRIPTION
**Pull request by mabraham**
This targets the oneAPI SYCL compiler with oneMKL as FFT implementation library.

### CODE REVIEW

**Comment by mkstoyanov** [comment]
Looks good. Thanks for contributing!

**Review by mkstoyanov — APPROVED** [review]

**Review by alecbcs — APPROVED** [review]
Looks good to me. @mabraham thanks for the new variant and nicely formatted additions!

File: var/spack/repos/builtin/packages/heffte/package.py [hunk 1]
```diff
@@ -68,6 +74,8 @@ class Heffte(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("rocsparse@3.8:", when="+magma+rocm", type=("build", "run"))
     depends_on("hipblas@3.8:", when="+magma+rocm", type=("build", "run"))
     depends_on("hipsparse@3.8:", when="+magma+rocm", type=("build", "run"))
+    depends_on("intel-oneapi-mkl@2023.2.0:", when="+sycl", type=("build", "run"))
```
Comment (rscohn2) [inline var/spack/repos/builtin/packages/heffte/package.py:77]: Do you recall why you are using explicit versions for mkl and intel mpi? We would like to move it forward.

Comment (mkstoyanov) [reply var/spack/repos/builtin/packages/heffte/package.py:77]: What do you mean by "explicit version"?

The `:` at the end imply "this version of newer", same as the ROCm libraries above.

Comment (rscohn2) [reply var/spack/repos/builtin/packages/heffte/package.py:77]: Sorry. I did not notice the ":"

**Pull request merged**